libvast/src/table_slice.cpp

```diff
@@ -706,9 +712,7 @@ filter(const table_slice& slice, const expression& expr, const ids& hints) {
   VAST_ASSERT(builder);
   auto flat_layout = flatten(slice.layout());
   auto check = [&](row_evaluator eval) {
-    if (expr == expression{})
-      return true;
-    return caf::visit(eval, expr);
+    return caf::visit(eval, *tailored_expr);
```

Comment:
Isn't this inverting the logic for the case of an empty expression? As far as I understand we used to return everything and now we return nothing. (although the tests seem to pass, so I'm probably misreading this somehow)

Reply:
Yep, and that's why the integration test is failing as well. Working on a fix rn.